### DISCUSSION TITLE
BLE temperature sensor with "Faikin Auto Mode" in either Heat or Cool (and not Auto Mode and locking to heat or cool)

### DISCUSSION THREAD

**Discussion by turbolooser** [revk/xESP32-Faikout#198] [Ideas]
As far as I do understand the dependency on the BLE temp sensor correctly its temperature will be only considered in Faikins Auto Mode right?

If I now run a Multi-Split environment with e.g. 3 IndoorUnits and only 1 of them has 1 Faikin Module I might run into conflicting modes on my Indoor Units with the "Faikin Auto Mode".

Could you implement that the Auto Mode (external BLE temp sensor) is either in Heat or Cool Mode and not Auto?

Would be perfect if I could just select that I would like to always use the external BLE sensor for temperature reference in any mode that the AC is running. To me the mode would be either heat/cool/auto which should take the BLE into account.

Thank you so much for your hard work

**Comment by revk** [comment]
There is a setting `lockmode` to stop if flipping mode, so it stays in heat or cool as you have set.

**Reply by turbolooser** [reply]
I know but thats not doing the trick with multiple units if you select "auto" as mode for the Faikin unit.
When I am switching the faikin unit from off to on in "auto" internally its set to "cool" like this
![image](https://github.com/revk/ESP32-Faikin/assets/115738363/a2ee5369-489a-4689-8af1-b20142926cb6)

Any other unit now trying to go into mode "heat" will fail as the mode can only be cool as the faikin unit did set it.

My workaround for now is to only set either "heat" or "cool" as mode for the faikin unit as internally the faikin then handles the "auto" but all other units do not seem to conflict then again.

Its a little tricky with multi-splits but I am ok with removing the "auto" from the HA mode selection as faikin seems to handle this internally then still correct

**Reply by revk** [reply]
The "faikin auto" is only "heat or cool" and it picks when you start based on how hot or cold it is. I can probably make it still switch even in `lockmode` if it is blocked by another unit in the opposite mode, if that helps?

**Reply by turbolooser** [reply]
I am not sure that it correctly picksup the intial mode based on the given temps as my target temp was far away but it was still initiating in "cool" but anyway.... whatever you think would be helpful is highly appreciated.
Awesome work

**Reply by revk** [reply]
Today has been mental, so will look tomorrow.

**Reply by revk** [reply]
Ok will make a change in next release, later today.

**Reply by turbolooser** [reply]
Did you implement that change already? Just to confirm 👍

**Reply by MUN0X** [reply]
I would like to know as well. It would be nice that, when lock mode is enabled, you could manually select heat or cool besides auto.

**Reply by revk** [reply]
In lock mode it uses whatever heat or cool most you have selected already and does not change it.

The change here is for when some units are hot and some are cold then it is allowed to change it if it is not able to do the mode it is in.

**Reply by MUN0X** [reply]
Yes but the thing is that in home assistant you can only select auto when the faikin-auto mode is enabled. Selecting any mode other than auto doesnt work.

Home assistant also doesnt report what the automode is doing (Heat or cooling) it just says auto. So you dont know what mode the unit is in to begin with.

if a user enables a unit and then changes the setpoint it might be in the wrong mode without the user knowing. That unit will then prevent other units that are being switched on from doing their thing due to the interlock. Allowing the units to switch to a different mode wont help. They are in the correct mode, they just cant use it.

It would therefore be nice if home assistant reports the active mode instead of just auto. And when lock mode is enabled, allow the user to select either heat, cool or auto just like the daikin itself does but maintain the benefits of the faikin-auto mode.

**Reply by revk** [reply]
Ok the reporting auto to HA was requested by people. So all a bit confusing. Not sure what is best approach.

**Reply by turbolooser** [reply]
@MUN0X 
I can set the faikin to "heat" and then it will internally switch to "auto" so thats working for me fine. I am using simple thermostat but I also checked this with the given default thermostat. Also using the "heat" option in a scheduler works fine for me... so I set to "heat" and faikin switches it automatically to "auto" without issues.
On simple thermostat I also do see the actual mode of operation (being heat) so I am fine on how this works right now.

**Comment by revk** [comment]
OK, this is HA specific it seems.

So, I have put the logic back so `lockmode` does not change mode, even if disabled by another unit in opposite mode.

I have made it so the mode is reported as `heat` or `cool` not `auto` if `lockmode` is set.

New code issue in a few minutes.

**Reply by MUN0X** [reply]
I don’t see the behaviour you describe. Still reverts to auto when selecting heat or cool. And only reports auto when the ac is on

**Reply by turbolooser** [reply]
same here I assume. I am sure it was showing "heat" earlier when setting "heat" (and internally it was going to "auto") but I cannot reproduce

**Reply by revk** [reply]
Ok and definitely on latest code?

**Reply by MUN0X** [reply]
Yes, definitely on latest stable version

**Reply by revk** [reply]
Ah, I think I see it. OK.

1. Upgrade s/w.
2. Once done, you have a new option to switch to *beta* software, change settings to that.
3. Upgrade s/w again.

That should address it. I'll include in next full release, and you can then change to not be *beta* any more.
